satellite/satellitedb/storagenodeaccounting.go

```diff
@@ -168,19 +173,54 @@ func (db *StoragenodeAccounting) SaveRollup(ctx context.Context, latestRollup ti
 				if err != nil {
 					return err
 				}
+
+				ts := getNodeTimestampName(ar.NodeID, accounting.LastRollup)
+				err = tx.UpdateNoReturn_AccountingTimestamps_By_Name(ctx,
+					dbx.AccountingTimestamps_Name(ts),
+					dbx.AccountingTimestamps_Update_Fields{
+						Value: dbx.AccountingTimestamps_Value(latestRollup),
+					},
+				)
+				if err != nil {
+					return err
+				}
 			}
 		}
 
-		return tx.UpdateNoReturn_AccountingTimestamps_By_Name(ctx,
-			dbx.AccountingTimestamps_Name(accounting.LastRollup),
-			dbx.AccountingTimestamps_Update_Fields{
-				Value: dbx.AccountingTimestamps_Value(latestRollup),
-			},
-		)
+		return nil
 	})
 	return Error.Wrap(err)
 }
 
+// NodeLastTimestamp returns the last tallied time by node, or nil.
+func (db *StoragenodeAccounting) NodeLastTimestamp(ctx context.Context, nodeID storj.NodeID, timestampType string) (_ time.Time, err error) {
+	defer mon.Task()(&ctx)(&err)
+	ts := getNodeTimestampName(nodeID, timestampType)
+	lastTally := time.Time{}
+	rowValue, err := db.db.Find_AccountingTimestamps_Value_By_Name(ctx, dbx.AccountingTimestamps_Name(ts))
+	if err != nil {
+		return lastTally, err
+	}
+	if rowValue == nil {
+		err = db.db.WithTx(ctx, func(ctx context.Context, tx *dbx.Tx) error {
+			lt, err := tx.Find_AccountingTimestamps_Value_By_Name(ctx, dbx.AccountingTimestamps_Name(ts))
+			if lt == nil {
+				return tx.CreateNoReturn_AccountingTimestamps(ctx,
+					dbx.AccountingTimestamps_Name(ts),
+					dbx.AccountingTimestamps_Value(lastTally),
```

Comment:
this is dropping a zero value into here. do we ever update it to be something other than the zero value? where do we update the node last timestamp?